src/dosbox.cpp

```diff
@@ -520,6 +520,15 @@ void DOSBOX_Init()
 	LOG_StartUp();
 #endif
 
+	pbool = secprop->Add_bool("pcjr_expanded_memory", only_at_start, true);
+	pbool->Set_help(
+		"Use PCjr expanded memory layout\n"
+		"  true:  640KB total memory with applications residing above 128KB.\n"
+		"         Compatible with most games. Required if a game needs more than 128KB.\n"
+		"  false: 128KB total memory with applications residing below 96KB\n"
+		"         Required for some older games which aren't compatible with PCjr memory upgrades"
+	);
+
```

Comment:
Nice description 🎉 A couple of suggestions and comments:

- `true` and `false` for toggling options is not very user-friendly for non-coders. So I introduced an enhancement for handling booleans a while ago: `on`, `enabled`, `yes`, `true` or `1` can be used for turning a boolean setting on, and `off`, `disabled`, `no`, `false` or `0` for turning it off.
- Consequently, I've eradicated the use of `true` and `false` from the config descriptions to use `on` and `off` instead.
- We always mention the default value on the first line, so this should be `Use PCjr expanded memory layout (enabled by default).`
- Space before `KB` (so `640KB` should become `640 KB`, etc.)
- I'd mention Jumpan and Troll for the standard memory layout option (just chuck `(e.g., Jumpan, Troll).` at the end).
- Periods at the ends of the bullet points (for consistency with the rest of the config descriptions).
- The description text needs to be wrapped to 79 characters so it looks nicely indented when displayed with `config -h <settingname>`. The last line goes above that limit:

<img width="1096" alt="image" src="https://github.com/dosbox-staging/dosbox-staging/assets/698770/61db4210-b6de-49c7-b1df-0e838da87df8">

---

...but with all said, I'd like to suggest not to go with a simple boolean value, but use an enum instead.

I 100% agree that we should keep the number of PCjr memory layout options to the minimum, but using a boolean paints us into a corner. If we need to add a third option in the future, we'd need to rename and deprecate the setting.

A slightly different approach open to potential future extension would be to call this `pcjr_memory_config` instead with only two options for now: `standard` and `expanded`, with the latter being the default.

Reply:
Yeah, good call on making it a string.  I wasn't thinking about keeping the config backwards compatible if we need to add another option later.  Changed + fixed up the help text.

Reply:
> Yeah, good call on making it a string. I wasn't thinking about keeping the config backwards compatible if we need to add another option later. Changed + fixed up the help text.

No problem. I've been doing the rename/deprecation dance of config settings for a while now, so I'm sensitive to it and prefer to avoid it if possible 😄